get-all-systems-with-domains.py

```diff
@@ -0,0 +1,283 @@
+
+        # If we still have a page value, we need to retrieve the piece to get to
+        # the next page.
+        if page is not None:
+            uri_parts = page["href"].split("/")
+            page = "{0}/{1}".format(uri_parts[-2], uri_parts[-1])
+            logger.info("...Processing next page...")
+        else:
+            logger.info("...Finished processing {0} records ({1} skipped).".format(
+                len(service_instances),
+                resources_skipped
+            ))
+
+    return service_instances
+
+
+def parse_resource(resource):
+    """
+    Takes in a CF API service instance resource and parses the following
+    information out of it:
+
+    - The broker name that manages the service instance
+    - The service plan name that was used with the broker
+    - The name associated with the service instance
+    - The created at date of the service instance
+    - The space name that the service instance resides in
+    - The organization name that the service instance resides in
+    """
+
+    logger.info("    ...Parsing {0} ({1})...".format(
+        resource["name"],
+        resource["created_at"].split("T")[0]
+    ))
+
+    # Start by pulling out the service instance information directly.
+    resource_info = {}
+    resource_info["instance_name"] = resource["name"]
+    resource_info["created_at"] = resource["created_at"].split("T")[0]
+
+    # Retrieve the service plan information.
+    service_plan_guid = resource["relationships"]["service_plan"]["data"]["guid"]
+    service_plan_output = json.loads(subprocess.check_output(["cf", "curl", CF_SERVICE_PLANS_API_URI + service_plan_guid]).decode("utf-8"))
+    resource_info["service_plan"] = service_plan_output["name"]
+
+    # Retrieve the broker information.
+    service_offering_guid = service_plan_output["relationships"]["service_offering"]["data"]["guid"]
+    service_offering_output = json.loads(subprocess.check_output(["cf", "curl", CF_SERVICE_OFFERINGS_API_URI + service_offering_guid]).decode("utf-8"))
+    resource_info["broker"] = service_offering_output["name"]
+
+    # Retrieve the space information.
+    space_guid = resource["relationships"]["space"]["data"]["guid"]
+    space_output = json.loads(subprocess.check_output(["cf", "curl", CF_SPACES_API_URI + space_guid]).decode("utf-8"))
+    resource_info["space_name"] = space_output["name"]
+
+    # Retrieve the organization information.
+    org_guid = space_output["relationships"]["organization"]["data"]["guid"]
+    org_output = json.loads(subprocess.check_output(["cf", "curl", CF_ORGANIZATIONS_API_URI + org_guid]).decode("utf-8"))
+    resource_info["org_name"] = org_output["name"]
+
+    return resource_info
+
+
+def get_fiscal_year(date_string):
+    """
+    Figures out and returns the fiscal year label based on a provided calendar
+    date.
+    """
+
+    year, month, day = date_string.split("-")
+    new_fiscal_year_months = ["10", "11", "12"]
+    fiscal_year = int(year[2:])
```

Comment:
well this is gonna blow up in 79 years /s

Reply:
I was about to follow up with another sarcastic remark, but watch us eat our words and this still be used 79 years from now...